cmd/cluster-controller/main.go

```diff
@@ -135,15 +135,7 @@ func main() {
 
 	go apiImporter.Start(ctx)
 	go apiresource.Start(ctx, apiResourceOptions.NumThreads)
-
-	if syncer != nil {
-		prepared, err := syncer.Prepare()
-		if err != nil {
-			fmt.Fprintln(os.Stderr, err)
-			os.Exit(1)
-		}
-		go prepared.Start(ctx)
-	}
+	go syncer.Start(ctx)
```

Comment:
I think you'll find that you still need the nil check here. If a syncer mode isn't specified, no syncer is not intended to run.

Reply:
sigh. Constructors returning nil.

Reply:
fixed